docs/common/object.ldoc

```diff
@@ -0,0 +1,31 @@
+
+--- Connect to a signal weakly. This allows the callback function to be garbage
+-- collected and automatically disconnects the signal when that happens.
+-- @tparam string name The name of the signal
+-- @tparam function func The callback to call when the signal is emitted
+-- @function weak_connect_signal
+
+--- Disonnect to a signal.
+-- @tparam string name The name of the signal
+-- @tparam function func The callback that should be disconnected
+-- @see add_signal
+-- @function disconnect_signal
+
+--- Emit a signal.
+--
+-- @tparam string name The name of the signal
+-- @param ... Extra arguments for the callback functions. Each connected
+--   function receives the object as first argument and then any extra arguments
+--   that are given to emit_signal()
+-- @function emit_signal
+
+--- Add a signal to an object. All signals must be added before they can be used.
+--
+-- @tparam string name The name of the new signal.
+-- @function add_signal
+
+--- Connect to a signal.
+-- @tparam string name The name of the signal
+-- @tparam function func The callback to call when the signal is emitted
+-- @see add_signal 
+-- @function connect_signal
```

Comment:
Why is this so far away from its weak version? I'd keep them closer together in this file.